From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fukiame <fukiame@proton.me>
Date: Tue, 4 Aug 2026 02:27:04 +0700
Subject: [PATCH] Revert "Refactor equalizer API to use EqBandSpec instead of
 raw gain values"

This reverts commit 1db7af37469215c5f0e13ba32825dbe690afdf4f.

diff --git a/lib/src/rust/frb_generated.dart b/lib/src/rust/frb_generated.dart
index f608bbeb..66f9956c 100644
--- a/lib/src/rust/frb_generated.dart
+++ b/lib/src/rust/frb_generated.dart
@@ -14,7 +14,6 @@ import 'api/uac2_api.dart';
 import 'audio/crossfader.dart';
 import 'audio/dsd_engine/dsd.dart';
 import 'audio/engine.dart';
-import 'audio/equalizer.dart';
 import 'dart:async';
 import 'dart:convert';
 import 'frb_generated.dart';
@@ -225,7 +224,7 @@ abstract class RustLibApi extends BaseApi {
 
   Future<void> crateApiAudioApiAudioSetEqualizer({
     required bool enabled,
-    required List<EqBandSpec> specs,
+    required List<double> gainsDb,
   });
 
   Future<void> crateApiAudioApiAudioSetFx({
@@ -1683,14 +1682,14 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
   @override
   Future<void> crateApiAudioApiAudioSetEqualizer({
     required bool enabled,
-    required List<EqBandSpec> specs,
+    required List<double> gainsDb,
   }) {
     return handler.executeNormal(
       NormalTask(
         callFfi: (port_) {
           final serializer = SseSerializer(generalizedFrbRustBinding);
           sse_encode_bool(enabled, serializer);
-          sse_encode_list_eq_band_spec(specs, serializer);
+          sse_encode_list_prim_f_32_loose(gainsDb, serializer);
           pdeCallFfi(
             generalizedFrbRustBinding,
             serializer,
@@ -1703,7 +1702,7 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
           decodeErrorData: sse_decode_String,
         ),
         constMeta: kCrateApiAudioApiAudioSetEqualizerConstMeta,
-        argValues: [enabled, specs],
+        argValues: [enabled, gainsDb],
         apiImpl: this,
       ),
     );
@@ -1712,7 +1711,7 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
   TaskConstMeta get kCrateApiAudioApiAudioSetEqualizerConstMeta =>
       const TaskConstMeta(
         debugName: "audio_set_equalizer",
-        argNames: ["enabled", "specs"],
+        argNames: ["enabled", "gainsDb"],
       );
 
   @override
@@ -3686,26 +3685,6 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
     return DsdRate.values[raw as int];
   }
 
-  @protected
-  EqBandSpec dco_decode_eq_band_spec(dynamic raw) {
-    // Codec=Dco (DartCObject based), see doc to use other codecs
-    final arr = raw as List<dynamic>;
-    if (arr.length != 4)
-      throw Exception('unexpected arr length: expect 4 but see ${arr.length}');
-    return EqBandSpec(
-      bandType: dco_decode_eq_band_type(arr[0]),
-      freqHz: dco_decode_f_32(arr[1]),
-      gainDb: dco_decode_f_32(arr[2]),
-      q: dco_decode_f_32(arr[3]),
-    );
-  }
-
-  @protected
-  EqBandType dco_decode_eq_band_type(dynamic raw) {
-    // Codec=Dco (DartCObject based), see doc to use other codecs
-    return EqBandType.values[raw as int];
-  }
-
   @protected
   double dco_decode_f_32(dynamic raw) {
     // Codec=Dco (DartCObject based), see doc to use other codecs
@@ -3751,9 +3730,9 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
   }
 
   @protected
-  List<EqBandSpec> dco_decode_list_eq_band_spec(dynamic raw) {
+  List<double> dco_decode_list_prim_f_32_loose(dynamic raw) {
     // Codec=Dco (DartCObject based), see doc to use other codecs
-    return (raw as List<dynamic>).map(dco_decode_eq_band_spec).toList();
+    return raw as List<double>;
   }
 
   @protected
@@ -4554,28 +4533,6 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
     return DsdRate.values[inner];
   }
 
-  @protected
-  EqBandSpec sse_decode_eq_band_spec(SseDeserializer deserializer) {
-    // Codec=Sse (Serialization based), see doc to use other codecs
-    var var_bandType = sse_decode_eq_band_type(deserializer);
-    var var_freqHz = sse_decode_f_32(deserializer);
-    var var_gainDb = sse_decode_f_32(deserializer);
-    var var_q = sse_decode_f_32(deserializer);
-    return EqBandSpec(
-      bandType: var_bandType,
-      freqHz: var_freqHz,
-      gainDb: var_gainDb,
-      q: var_q,
-    );
-  }
-
-  @protected
-  EqBandType sse_decode_eq_band_type(SseDeserializer deserializer) {
-    // Codec=Sse (Serialization based), see doc to use other codecs
-    var inner = sse_decode_i_32(deserializer);
-    return EqBandType.values[inner];
-  }
-
   @protected
   double sse_decode_f_32(SseDeserializer deserializer) {
     // Codec=Sse (Serialization based), see doc to use other codecs
@@ -4641,15 +4598,10 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
   }
 
   @protected
-  List<EqBandSpec> sse_decode_list_eq_band_spec(SseDeserializer deserializer) {
+  List<double> sse_decode_list_prim_f_32_loose(SseDeserializer deserializer) {
     // Codec=Sse (Serialization based), see doc to use other codecs
-
     var len_ = sse_decode_i_32(deserializer);
-    var ans_ = <EqBandSpec>[];
-    for (var idx_ = 0; idx_ < len_; ++idx_) {
-      ans_.add(sse_decode_eq_band_spec(deserializer));
-    }
-    return ans_;
+    return deserializer.buffer.getFloat32List(len_);
   }
 
   @protected
@@ -5527,21 +5479,6 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
     sse_encode_i_32(self.index, serializer);
   }
 
-  @protected
-  void sse_encode_eq_band_spec(EqBandSpec self, SseSerializer serializer) {
-    // Codec=Sse (Serialization based), see doc to use other codecs
-    sse_encode_eq_band_type(self.bandType, serializer);
-    sse_encode_f_32(self.freqHz, serializer);
-    sse_encode_f_32(self.gainDb, serializer);
-    sse_encode_f_32(self.q, serializer);
-  }
-
-  @protected
-  void sse_encode_eq_band_type(EqBandType self, SseSerializer serializer) {
-    // Codec=Sse (Serialization based), see doc to use other codecs
-    sse_encode_i_32(self.index, serializer);
-  }
-
   @protected
   void sse_encode_f_32(double self, SseSerializer serializer) {
     // Codec=Sse (Serialization based), see doc to use other codecs
@@ -5600,15 +5537,15 @@ class RustLibApiImpl extends RustLibApiImplPlatform implements RustLibApi {
   }
 
   @protected
-  void sse_encode_list_eq_band_spec(
-    List<EqBandSpec> self,
+  void sse_encode_list_prim_f_32_loose(
+    List<double> self,
     SseSerializer serializer,
   ) {
     // Codec=Sse (Serialization based), see doc to use other codecs
     sse_encode_i_32(self.length, serializer);
-    for (final item in self) {
-      sse_encode_eq_band_spec(item, serializer);
-    }
+    serializer.buffer.putFloat32List(
+      self is Float32List ? self : Float32List.fromList(self),
+    );
   }
 
   @protected
-- 
2.55.0

